packages/@aws-cdk/aws-ec2/lib/vpn.ts

```diff
@@ -0,0 +1,123 @@
+import cdk = require('@aws-cdk/cdk');
+import { CfnCustomerGateway, CfnVPNConnection, CfnVPNConnectionRoute } from './ec2.generated';
+import { IVpcNetwork } from './vpc-ref';
+
+export interface IVpnConnection extends cdk.IConstruct {
+  /**
+   * The id of the VPN connection.
+   */
+  readonly vpnId: string;
+
+  /**
+   * The id of the customer gateway.
+   */
+  readonly customerGatewayId: string;
+
+  /**
+   * The ip address of the customer gateway.
+   */
+  readonly customerGatewayIp: string;
+
+  /**
+   * The ASN of the customer gateway.
+   */
+  readonly customerGatewayAsn: number;
+}
+export interface VpnTunnelOption {
+  /**
+   * The pre-shared key (PSK) to establish initial authentication between the virtual
+   * private gateway and customer gateway.
+   */
+  presharedKey: string;
+
+  /**
+   * The range of inside IP addresses for the tunnel. Any specified CIDR blocks must be
+   * unique across all VPN connections that use the same virtual private gateway.
+   */
+  tunnelInsideCidr: string;
+}
+
+export interface BaseVpnConnectionProps {
+  /**
+   * The ip address of the customer gateway.
+   */
+  ip: string;
+
+  /**
+   * The ASN of the customer gateway.
+   *
+   * @default 65000
+   */
+  asn?: number;
+
+  /**
+   * The static routes to be routed from the VPN gateway to the customer gateway.
+   *
+   * @default Dynamic routing (BGP)
+   */
+  staticRoutes?: string[];
+
+  /**
+   * Tunnel options for the VPN connection.
+   */
+  vpnTunnelOptions?: VpnTunnelOption[];
+}
+
+export interface VpnConnectionProps extends BaseVpnConnectionProps {
+  /**
+   * The VPC to connect to.
+   */
+  vpc: IVpcNetwork;
+}
+
+/**
+ * The IPsec 1 VPN connection type.
+ */
+export const IPsec1 = 'ipsec.1';
```

Comment:
Should this be an enum maybe?

Reply:
Yes

Reply:
Switching to an enum with only one member seems problematic for `jsii`? See build logs...

Reply:
You are correct: https://github.com/awslabs/jsii/issues/231

Can you add another member, even a dummy one?

Reply:
And you can add this to the jsii issue and we'll fix once the jsii bug is fixed.